src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -659,7 +660,7 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
                 {
                     countOfExplicitNames++;
 
-                    if (TupleTypeSymbol.IsMemberNameReserved(name) && name != TupleTypeSymbol.UnderlyingMemberName(i))
+                    if (TupleTypeSymbol.IsMemberNameReserved(name) && name != TupleTypeSymbol.TupleMemberName(i))
```

Comment:
It feels like we should reserve "Rest" as a name at any position. In addition we can simplify this check and avoid unnecessary string concatenations and comparisons if we modify IsMemberNameReserved to return the index, at which reserved name can be used, -1 if not reserved. Then ```name != TupleTypeSymbol.TupleMemberName(i)``` condition can be simplified to ```<reserved index> != i```.